code/game/mecha/mecha.dm

```diff
@@ -565,6 +565,8 @@
 
 	if(prob(deflect_chance * deflection) && (Proj.damage_type == BRUTE || Proj.damage_type == BURN))

```

Comment:
why not just move the `BRUTE` and `BURN` damage checks out, independently, so you don't have to do specific checks for every single damage type under the sun? There's more than just `STAMINA` that you'd technically have to correct for, here; checking for them all would be messy.